### PR TITLE
[WIP] provider/aws: Add aws_ecs_container_definitions

### PR DESCRIPTION
**Pull request by jen20**
This is a work in progress commit to add a data source similar in nature to `aws_iam_policy_document` which can be used for describing the containers in an ECS task.

### CODE REVIEW

File: builtin/providers/aws/data_source_aws_ecs_container_definitions_model.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package aws
+
+type ecsContainerDefinition struct {
```
Comment (radeksimko) [inline builtin/providers/aws/data_source_aws_ecs_container_definitions_model.go]: Is there any specific reason we can't use the official structs from the SDK? https://github.com/aws/aws-sdk-go/blob/master/service/ecs/api.go#L3135

Comment (jen20) [inline builtin/providers/aws/data_source_aws_ecs_container_definitions_model.go]: No, I never thought to look for them. I'll update this to use them instead!

**Comment by jen20** [comment]
@radeksimko, @stack72 - looking more at this it might be better to just add this structure to the existing task definition and accept it as a breaking change (having accepted JSON before). Any thoughts on that?

**Comment by radeksimko** [comment]
@jen20 I'd personally try to avoid deeply nested and complex schemas and the task definition schema has (IMO) a potential to grow, so I'm not inclined to that idea, especially after I spent this week going through schema of `kubernetes_pod` which I'm working on.

I however agree that asking for raw JSON is not very idiomatic, so having such data source would be very helpful and we should encourage people to use that instead of JSON.

As a side note I feel like Terraform would benefit from having a separate 1st class name/place for these "data sources" which aren't really making any API calls and just help users to build complex configs.

**Comment by apparentlymart** [comment]
As part of the the Terraform 0.10 release earlier this year, all of the Terraform providers were moved to their own repositories in [the terraform-providers GitHub organization](https://github.com/terraform-providers), and removed from the Terraform Core repository.

Unfortunately due to the fact that new issues and pull requests are being opened constantly, it was not possible for the various provider maintainers to merge all outstanding pull requests before this split, and there is no automatic way to migrate a pull request to a new repository.

As a result, this pull request can sadly no longer be applied as-is, and so I'm going to close it.

If someone has the time and motivation to apply same changes to [the `aws` provider repository](https://github.com/terraform-providers/terraform-provider-aws) and open a new PR there, the maintainers of that provider should be able to review and merge it.

Thanks again for working on this, and sorry it was not able to be merged before the provider repository changes.

**Pull request closed without merging**